src/Text/Pretty/Simple/Internal/Expr.hs

```diff
@@ -43,5 +45,6 @@ data Expr
   -- only thing we will be doing with it is turning it /back/ into a string
   -- at some stage, so we might as well cut out the middle man and store it
   -- directly like this.
+  | CustomExpr Style !String
```

Comment:
This opens up a lot of flexibility. But it feels slightly ugly, since we will never actually parse a `CustomExpr`.  We could formalise that fact with GADTs and DataKinds, but it's possibly unwarranted complexity.